managed/services/management/backup/backups_service.go

```diff
@@ -229,7 +236,14 @@ func (s *BackupsService) ScheduleBackup(ctx context.Context, req *backupv1beta1.
 				return status.Errorf(codes.InvalidArgument, "Can't create mySQL backup task: %v", err)
 			}
 		case models.MongoDBServiceType:
-			backupParams.DataModel = models.LogicalDataModel
+			switch req.DataModel {
+			case backupv1beta1.DataModel_LOGICAL:
+				backupParams.DataModel = models.LogicalDataModel
+			case backupv1beta1.DataModel_PHYSICAL:
+				backupParams.DataModel = models.PhysicalDataModel
+			default:
+				return status.Errorf(codes.InvalidArgument, "invalid data model: %s", req.DataModel.String())
+			}
```

Comment:
We have same code on lines `117-124`, it seems better to extract it to some function.